cmd/sti/main.go

```diff
@@ -191,6 +192,56 @@ func newCmdBuild(cfg *api.Config) *cobra.Command {
 	return buildCmd
 }
 
+func newCmdRebuild(cfg *api.Config) *cobra.Command {
+	buildCmd := &cobra.Command{
+		Use:   "rebuild <image>",
+		Short: "Rebuild an existing image",
+		Long:  "Rebuild an existing application image, that was build by S2I previously.",
+		Run: func(cmd *cobra.Command, args []string) {
+			// If user specifies the arguments, then we override the stored ones
+			if len(args) >= 0 {
+				cfg.Tag = args[0]
+			}
+
+			if r, err := os.Open(cfg.DockerCfgPath); err == nil {
+				cfg.PullAuthentication = docker.GetImageRegistryAuth(r, cfg.Tag)
+			}
+
+			err := build.GenerateConfigFromLabels(cfg.Tag, cfg)
+			checkErr(err)
+
+			// Attempt to read the .dockercfg and extract the authentication for
+			// docker pull
+			if r, err := os.Open(cfg.DockerCfgPath); err == nil {
+				cfg.PullAuthentication = docker.GetImageRegistryAuth(r, cfg.BuilderImage)
+			}
+
+			if glog.V(2) {
+				fmt.Printf("\n%s\n", describe.DescribeConfig(cfg))
+			}
+
+			builder, err := strategies.GetStrategy(cfg)
+			checkErr(err)
+			result, err := builder.Build(cfg)
+			checkErr(err)
+
+			for _, message := range result.Messages {
+				glog.V(1).Infof(message)
+			}
+
+		},
+	}
+
+	buildCmd.Flags().BoolVarP(&(cfg.Quiet), "quiet", "q", false, "Operate quietly. Suppress all non-error output.")
+	buildCmd.Flags().BoolVar(&(cfg.Incremental), "incremental", true, "Perform an incremental build")
+	buildCmd.Flags().BoolVar(&(cfg.RemovePreviousImage), "rm", false, "Remove the previous image during incremental builds")
+	buildCmd.Flags().StringVar(&(cfg.CallbackURL), "callback-url", "", "Specify a URL to invoke via HTTP POST upon build completion")
+	buildCmd.Flags().BoolVar(&(cfg.ForcePull), "force-pull", true, "Always pull the builder image even if it is present locally")
```

Comment:
@bparees just checked the GenerateConfigFromLabels() method in pkg/util/labels.go

The method will do CheckAndPull() on the application image you provided. What this method does it that it checks if the image is available locally and if not it will pull it from registry.